src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.Core.DataStore.Sql;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using Dfc.CourseDirectory.Core.Models;
+using MediatR;
+using OneOf;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.FindACourseApi.Features.TLevelDetail
+{
+    public class Query : IRequest<OneOf<NotFound, TLevelDetailViewModel>>
+    {
+        public Guid TLevelId { get; set; }
+    }
+
+    public class Handler : IRequestHandler<Query, OneOf<NotFound, TLevelDetailViewModel>>
+    {
+        private readonly ISqlQueryDispatcher _sqlQueryDispatcher;
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+
+        public Handler(ISqlQueryDispatcher sqlQueryDispatcher, ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        {
+            _sqlQueryDispatcher = sqlQueryDispatcher;
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+        }
+
+        public async Task<OneOf<NotFound, TLevelDetailViewModel>> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var tLevel = await _sqlQueryDispatcher.ExecuteQuery(new GetTLevel() { TLevelId = request.TLevelId });
+
+            if (tLevel == null)
+            {
+                return new NotFound();
+            }
+
+            var getProvider = _cosmosDbQueryDispatcher.ExecuteQuery(
```

Comment:
Why do we need to use the Cosmos Provider collection? Is everything not in SQL?

Reply:
Mainly for symmetry with the CourseRunDetail one. The current SQL queries don't retrieve `Contacts`, which is required here. In this interest of getting this out of the door quickly I did the simplest thing.